cfn-resources/federated-query-limit/cmd/resource/resource.go

```diff
@@ -0,0 +1,334 @@
+
+	modelValidation := validator.ValidateModel(ListRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	// Create atlas client
+	if currentModel.Profile == nil || *currentModel.Profile == "" {
+		currentModel.Profile = aws.String(profile.DefaultProfile)
+	}
+
+	atlas, peErr := util.NewAtlasClient(&req, currentModel.Profile)
+	if peErr != nil {
+		return *peErr, nil
+	}
+	listQueryLimitsAPIRequest := atlas.AtlasV2.DataFederationApi.ReturnFederatedDatabaseQueryLimits(
+		context.Background(),
+		*currentModel.ProjectId,
+		*currentModel.TenantName,
+	)
+
+	listQueryLimitsAPIResult, response, err := listQueryLimitsAPIRequest.Execute()
+
+	defer closeResponse(response)
+	if err != nil {
+		_, _ = logger.Warnf("Execute error: %s", err.Error())
+		return handleError(response, LIST, err)
+	}
+	queryLimits := make([]interface{}, 0)
+	for i := range listQueryLimitsAPIResult {
+		queryLimit := Model{
+			ProjectId: currentModel.ProjectId,
+			Profile:   currentModel.Profile,
+		}
+		queryLimit.getQueryLimit(&listQueryLimitsAPIResult[i])
+		queryLimits = append(queryLimits, queryLimit)
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "List Completed",
+		ResourceModels:  queryLimits}, nil
+}
+
+func closeResponse(response *http.Response) {
+	if response != nil {
+		response.Body.Close()
+	}
+}
+
+func handleError(response *http.Response, method string, err error) (handler.ProgressEvent, error) {
+	errMsg := fmt.Sprintf("%s error:%s", method, err.Error())
+	_, _ = logger.Warn(errMsg)
+	if response.StatusCode == http.StatusConflict {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          errMsg,
+			HandlerErrorCode: cloudformation.HandlerErrorCodeAlreadyExists}, nil
+	}
+	return progress_events.GetFailedEventByResponse(errMsg, response), nil
+}
+
+func getFederatedQueryLimit(atlas *util.MongoDBClient, currentModel *Model) (*atlasSDK.DataFederationTenantQueryLimit, *http.Response, error) {
+	getQueryLimitAPIRequest := atlas.AtlasV2.DataFederationApi.ReturnFederatedDatabaseQueryLimit(
+		context.Background(),
+		*currentModel.ProjectId,
+		*currentModel.TenantName,
+		*currentModel.LimitName,
+	)
+	queryLimit, response, err := getQueryLimitAPIRequest.Execute()
+	return queryLimit, response, err
+}
+
+func createOrUpdateQueryLimit(currentModel *Model, atlas *util.MongoDBClient, method string) (handler.ProgressEvent, error) {
+	queryLimitInput := currentModel.setQueryLimit()
+	createQueryLimitRequest := atlas.AtlasV2.DataFederationApi.CreateOneDataFederationQueryLimit(
+		context.Background(),
+		*currentModel.ProjectId,
+		*currentModel.TenantName,
+		*currentModel.LimitName,
+		queryLimitInput,
+	)
+	_, response, err := createQueryLimitRequest.Execute()
+	defer closeResponse(response)
+	if err != nil {
+		return handleError(response, method, err)
+	}
+
+	queryLimit, response, err := getFederatedQueryLimit(atlas, currentModel)
```

Comment:
here ther is a problem, you are calling to do the get again, first in the create to validate if it exists, and again here, is there a way to avoid making this second call? can we use the output from the create on line 274 to complete the model?

Reply:
Removed extra Get Call